.github/actions/chunk_test_suite/action.yml

```diff
@@ -0,0 +1,51 @@
+name: Determine optimal test chunks
+description: Chunk test suite
+inputs:
+  build-directory:
+    description: Path to the build directory
+    required: true
+  source-directory:
+    description: Path to the source directory
+    required: true
+  number-of-chunks:
+    description: Number of chunks
+    required: true
+  junit-report-artifact-name:
+    description: Name of the job that reports the junit summary
+    required: true
+outputs:
+  chunk-array:
+    description: Array of chunks
+    value: ${{ steps.extract-chunks.outputs.chunk_array }}
+runs:
+  using: composite
+  steps:
+    - name: Download junit test reports
+      id: download-junit-reports
+      continue-on-error: true
+      uses: dawidd6/action-download-artifact@v6
+      with:
+        workflow: nightly_tests.yml
+        name: ${{ inputs.junit-report-artifact-name }}
+        path: ${{ inputs.source-directory }}
+        branch: ${{ github.event.repository.default_branch }}
+        workflow_conclusion: success
+        if_no_artifact_found: fail
```

Comment:
Does this mean that we can never run PR actions if the nightly pipeline fails? This would lead to a deadlock. If this is indeed the case, we need a fallback for the chunks.

Reply:
no, we have `continue-on-error: true` which makes this action end with `success` even if the artifact was not found. I can use the state later with `steps.download-junit-reports.outcome`.

I use this to determine whether we found an artifact. If yes, we add `--junit-report report.xml` in the next step.

Take this as an example: https://github.com/4C-multiphysics/4C/actions/runs/12237519164/job/34174380024
The artifact was not available yesterday, the step itself failed (output: `Error: no artifacts found`), but the job succeeded.